File: Manufacturing/src/Python/package/microsoft_azure_sphere_deviceapi/capabilities.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from microsoft_azure_sphere_deviceapi import utils

__all__ = ['get_device_capabilities']


def get_device_capabilities() -> dict:
    """Makes a "GET" request to show the current device capability configuration of
    the attached device.

    :return: The device capability configuration on success. An exception will be thrown on error.
    :rtype: dict[str, str]
    :raises: requests.exceptions
    :raises: AzureSphereDeviceApiException
    """
    return utils.get_request("device/capabilities")

File: Manufacturing/src/CSharp/Nuget/Package/Capabilities.cs
/* Copyright (c) Microsoft Corporation. All rights reserved.
   Licensed under the MIT License. */
namespace Microsoft.Azure.Sphere.DeviceAPI
{
    /// <summary>
    /// Device  REST APIs to manage device capability configurations.
    /// </summary>
    public static class Capabilities
    {
        /// <summary>Makes a "GET" request to retrieve the current device capability configuration of the attached device.</summary>
        /// <returns>The device capability configuration as a string on success. An exception will be thrown on error.</returns>
        public static string GetDeviceCapabilities()
        {
            SinceDeviceAPIVersion.ValidateDeviceApiVersion("GetDeviceCapabilities");
            return RestUtils.GetRequest("device/capabilities");
        }
    }
}
